src/Sarif/Readers/LineMappingStreamReader.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Readers
+{
+    /// <summary>
+    ///  LineMappingStreamReader is a StreamReader which additionally tracks the byte offsets of each line start
+    ///  in the part of the file currently read. This allows it to map a line and char back to a byte offset
+    ///  to allow consumers to seek back to the position later.
+    /// </summary>
+    internal class LineMappingStreamReader : StreamReader
```

Comment:
>LineMappingStreamReader [](start = 19, length = 23)

Would you mind taking a look at NewlineIndex.cs to see whether we have convergence possibilities?